File: snippets/fsharp/VS_Snippets_CLR_System/system.console.windowleft/fs/windowleft1.fs
// <Snippet1>
open System

let showConsoleStatistics () =
    printfn "Console statistics:"
    printfn $"   Buffer: {Console.BufferHeight} x {Console.BufferWidth}" 
    printfn $"   Window: {Console.WindowHeight} x {Console.WindowWidth}"
    printfn $"   Window starts at {Console.WindowLeft}."
    printfn "Press <- or -> to move window, Ctrl+C to exit."

Console.BufferWidth <- Console.BufferWidth + 4
Console.Clear()

showConsoleStatistics ()

let mutable moved = false

while true do
    let key = Console.ReadKey true
    if key.Key = ConsoleKey.LeftArrow then
        let pos = Console.WindowLeft - 1
        if pos >= 0 && pos + Console.WindowWidth <= Console.BufferWidth then
            Console.WindowLeft <- pos
            moved <- true
    elif key.Key = ConsoleKey.RightArrow then
        let pos = Console.WindowLeft + 1
        if pos + Console.WindowWidth <= Console.BufferWidth then
            Console.WindowLeft <- pos
            moved <- true
    if moved then
        showConsoleStatistics ()
        moved <- false
    
    printfn ""

// </Snippet1>

File: snippets/csharp/System/Console/WindowLeft/windowleft1.cs
// <Snippet1>
using System;

public class Example
{
    public static void Main()
    {
        ConsoleKeyInfo key;
        bool moved = false;

        Console.BufferWidth += 4;
        Console.Clear();

        ShowConsoleStatistics();
        do
        {
            key = Console.ReadKey(true);
            if (key.Key == ConsoleKey.LeftArrow)
            {
                int pos = Console.WindowLeft - 1;
                if (pos >= 0 && pos + Console.WindowWidth <= Console.BufferWidth)
                {
                    Console.WindowLeft = pos;
                    moved = true;
                }
            }
            else if (key.Key == ConsoleKey.RightArrow)
            {
                int pos = Console.WindowLeft + 1;
                if (pos + Console.WindowWidth <= Console.BufferWidth)
                {
                    Console.WindowLeft = pos;
                    moved = true;
                }
            }
            if (moved)
            {
                ShowConsoleStatistics();
                moved = false;
            }
            Console.WriteLine();
        } while (true);
    }

    private static void ShowConsoleStatistics()
    {
        Console.WriteLine("Console statistics:");
        Console.WriteLine($"   Buffer: {Console.BufferHeight} x {Console.BufferWidth}");
        Console.WriteLine($"   Window: {Console.WindowHeight} x {Console.WindowWidth}");
        Console.WriteLine($"   Window starts at {Console.WindowLeft}.");
        Console.WriteLine("Press <- or -> to move window, Ctrl+C to exit.");
    }
}
// </Snippet1>
